addon/components/frost-toggle-button.js

```diff
@@ -0,0 +1,106 @@
+import Ember from 'ember'
+const {computed, Component, ViewUtils} = Ember
+import layout from '../templates/components/frost-toggle-button'
+import FrostEventsProxy from '../mixins/frost-events-proxy'
+
+export default Component.extend(FrostEventsProxy, {
+
+  // ==========================================================================
+  // Properties
+  // ==========================================================================
+
+  layout: layout,
+
+  classNames: ['frost-toggle-button'],
+
+  attributeBindings: [
+    'toggled',
+    'disabled'
+  ],
+
+  classNameBindings: [
+    'disabled'
+  ],
+
+  size: 'medium',
+  toggled: false,
+
+  // ==========================================================================
+  // Functions
+  // ==========================================================================
+
+  init () {
+    this._super(...arguments)
+
+    const toggled = this.toggled
+    Ember.assert(`The value passed to ${this.toString()}{toggled} must be boolean`,
+      typeof toggled === 'boolean' || 'undefined')
+    if (toggled) {
+      this.set('toggled', true)
+    }
+  },
+
+  // ==========================================================================
+  // Computed Properties
+  // ==========================================================================
+
+  // UX requirement - label need hold the value
+  _onLabel: computed('onLabel', function () {
+    const onLabel = this.get('onLabel')
+    if (onLabel && typeof onLabel === 'string') {
+      return onLabel
+    }
+    return true
+  }),
+
+  _offLabel: computed('offLabel', function () {
+    const offLabel = this.get('offLabel')
+    if (offLabel && typeof offLabel === 'string') {
+      return offLabel
+    }
+    return false
+  }),
+
+  _onValue: computed('onValue', '_onLabel', function () {
+    const onValue = this.get('onValue')
+    return onValue || this.get('_onLabel')
+  }),
+
+  _offValue: computed('offValue', '_offLabel', function () {
+    const offValue = this.get('offValue')
+    return offValue || this.get('_offLabel')
+  }),
+
+  _value: computed('onValue', 'offValue', 'toggled', function () {
```

Comment:
Is this necessary or can we handle the value calculation inside the click event?